sound/soc/sof/intel/pci-mtl.c

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: (GPL-2.0-only OR BSD-3-Clause)
+//
+// This file is provided under a dual BSD/GPLv2 license.  When using or
+// redistributing this file, you may do so under either license.
+//
+// Copyright(c) 2018-2022 Intel Corporation. All rights reserved.
+//
+// Author: Ranjani Sridharan <ranjani.sridharan@linux.intel.com>
+//
+
+#include <linux/module.h>
+#include <linux/pci.h>
+#include <sound/soc-acpi.h>
+#include <sound/soc-acpi-intel-match.h>
+#include <sound/sof.h>
+#include "../ops.h"
+#include "../sof-pci-dev.h"
+
+/* platform specific devices */
+#include "hda.h"
+#include "mtl.h"
+
+static const struct sof_dev_desc mtl_desc = {
+	.use_acpi_target_states	= true,
+	.machines               = snd_soc_acpi_intel_mtl_machines,
+	.alt_machines		= snd_soc_acpi_intel_mtl_sdw_machines,
+	.resindex_lpe_base      = 0,
+	.resindex_pcicfg_base   = -1,
+	.resindex_imr_base      = -1,
+	.irqindex_host_ipc      = -1,
+	.chip_info = &mtl_chip_info,
+	.ipc_supported_mask	= BIT(SOF_INTEL_IPC4),
+	.ipc_default		= SOF_INTEL_IPC4,
+	.default_fw_path = {
+		[SOF_INTEL_IPC4] = "intel/avs/mtl",
```

Comment:
we have to decide a new path for MTL, this shouldn't be 'avs' to avoid confusion with the closed-source firmware.

Reply:
> we have to decide a new path for MTL, this shouldn't be 'avs' to avoid confusion with the closed-source firmware.

How about csof (converged SOF)?

Reply:
can we just do it as "intel/sof/"

Reply:
> can we just do it as "intel/sof/"

If we want to use the same firmware folder as IPC3 firmware, we will need a new IPC4 firmware binary name. Besides, we use the same topology name for both IPC3 and IPC4, so we can't use the same topology folder.

Reply:
intel/sof-ipc4/mtl